src/pages/Browse/index.spec.tsx

```diff
@@ -607,6 +607,79 @@ describe(__filename, () => {
     expect(root.find('title')).toHaveText(`Browse ${name}: ${versionString}`);
   });
 
+  it('updates the selected path if it is different from the path in the URL', () => {
+    const versionId = nextUniqueId();
+    const newPath = 'new/path';
+    const initialPath = 'initial/path';
+    const history = createFakeHistory({
+      location: createFakeLocation({
+        search: queryString.stringify({ path: newPath }),
+      }),
+    });
+    const { renderAndUpdate } = setUpVersionFileUpdate({
+      initialPath,
+      versionId,
+    });
+
+    const { dispatchSpy } = renderAndUpdate({ history });
+
+    expect(dispatchSpy).toHaveBeenCalledWith(
+      versionsActions.updateSelectedPath({ selectedPath: newPath, versionId }),
+    );
+  });
+
+  it('updates the selected path to the initial path if the path is undefined in the URL', () => {
+    const versionId = nextUniqueId();
+    const initialPath = 'initial/path';
+    const history = createFakeHistory({
+      location: createFakeLocation({
+        search: '',
+      }),
+    });
+    const { renderAndUpdate, store } = setUpVersionFileUpdate({
+      initialPath,
+      versionId,
+    });
+    store.dispatch(
+      versionsActions.updateSelectedPath({
+        selectedPath: 'another-path',
+        versionId,
+      }),
+    );
+
+    const { dispatchSpy } = renderAndUpdate({ history });
+
+    expect(dispatchSpy).toHaveBeenCalledWith(
+      versionsActions.updateSelectedPath({
+        selectedPath: initialPath,
+        versionId,
+      }),
+    );
+  });
+
+  it('does not dispatch updateSelectedPath if selectedPath has been updated', () => {
```

Comment:
Should there also be a test for the case where `path` is undefined in the URL, and we have navigated back to the file that is for the `initialPath`, and which verifies that `updateSelectedPath` is not dispatched in this case as well?